conf.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/ini.v1"
+)
+
+const (
+	cliLogLevel  = "log-level"
+	cliCertFile  = "cert-file"
+	cliKeyFile   = "key-file"
+	cliAPIServer = "base_url"
+)
+
+type Conf struct {
+	CertFile  string
+	KeyFile   string
+	LogLevel  string
+	CADir     string
+	APIServer string
+}
+
+func (conf *Conf) CreateTLSClientConfig() (*tls.Config, error) {
+	var certData, keyData []byte
+	var err error
+	rootCAs := make([][]byte, 0)
+
+	KeyDir := conf.KeyFile
+	CertDir := conf.CertFile
+
+	certData, err = os.ReadFile(CertDir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read cert-file: %w", err)
+	}
+
+	keyData, err = os.ReadFile(KeyDir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read key-file: %w", err)
+	}
+	CAFiles, err := os.ReadDir(conf.CADir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read ca files: %w", err)
+	}
+	for _, file := range CAFiles {
+		fPath := filepath.Join(conf.CADir, file.Name())
+		data, err := os.ReadFile(fPath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read ca-file %s : %w", fPath, err)
+		}
+		rootCAs = append(rootCAs, data)
+	}
+
+	tlsConfig := &tls.Config{}
+	cert, err := tls.X509KeyPair(certData, keyData)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create key pair: %w", err)
+	}
+	tlsConfig.Certificates = []tls.Certificate{cert}
+
+	pool, err := x509.SystemCertPool()
```

Comment:
This is actually not correct. We do not want to load system CA certs. We only want to load CA certs from `conf.CADir`. Here is snapshot of code how to simplify:

```go
	caFiles, err := os.ReadDir(*caDir)
	if err != nil {
		return nil, fmt.Errorf("cannot read PEM files from CA directory: %w", err)
	}
	caCertPool := x509.NewCertPool()
	for _, file := range caFiles {
		fPath := filepath.Join(*caDir, file.Name())
		data, err := os.ReadFile(fPath)
		if err != nil {
			return nil, fmt.Errorf("cannot read CA PEM file %s : %w", fPath, err)
		}
		caCertPool.AppendCertsFromPEM(data)
	}
```